docs/faq.rst

```diff
@@ -241,3 +241,36 @@ Common gotchas to watch out for involve not ``free()``-ing memory region
 that that were ``malloc()``-ed in another shared library, using data
 structures with incompatible ABIs, and so on. pybind11 is very careful not
 to make these types of mistakes.
+
+Inconsistent detection of Python version in CMake and pybind11
+==============================================================
+
+The functions ``find_package(PythonInterp)`` and ``find_package(PythonLibs)`` provided by CMake
+are not consistent with detection of Python version in pybind11. If several versions of Python are installed
+in the Linux system (i.e. 2.7 and 3.5) the pybind11 defaults to 3.5, while CMake usually defaults to 2.7.
+The reason for this design decision is unreliable and buggy behavior of ``find_package(PythonInterp)`` and ``find_package(PythonLibs)``
+which are among the worst-designed parts of CMake. We don't want to mimick this behavior in pybind11 in any way.
```

Comment:
While I don't disagree with this statement, I do find the language a little strong for putting in the main documentation.  Here's a suggestion for these first two paragraphs (feel free to edit):

```
The functions ``find_package(PythonInterp)`` and ``find_package(PythonLibs)`` provided by CMake
for Python version detection are not used by pybind11 due to unreliability and limitations that make
them unsuitable for pybind11's needs.  Instead pybind provides its own, more reliable Python detection
CMake code.  Conflicts can arise, however, when using pybind11 in a project that *also* uses the CMake
Python detection.

Consider the following CMake code when run on a system with both Python 2.7 and 3.6 installed:
```